### ISSUE TITLE
Pass KatharsisInvokerContext to RequestDispatcher

### ISSUE THREAD

**Issue by Ramblurr** [katharsis-project/katharsis-framework#154]
The `RequestDispatcher` is one of the central bits in the request handling of Katharsis. It is easy to supply your own request dispatcher, which is great for extensibility. 

However inside the request dispatcher we don't have access to the request context nor the `KatharsisInvokerContext`. In my case I want to access to access the request headers for logging purposes.

Should we pass the invoker context into the request dispatcher?

**Comment by chb0github** [comment]
Without the invoker context it's impossible to do some things; like
accessing request headers. My vote is to pass it along

**Comment by Ramblurr** [comment]
Yup, access to headers is quite important IMHO. Otherwise we need a Katharsis Filter to do things w/ katharsis context, and another filter before Katharsis to do things with the headers.

**Comment by chb0github** [comment]
Yeah, that limitation made it nearly impossible (and indeed, based on
another bug, I never did get it right) to get the host header from the
request and thus create appropriate resource links.

If you're made this change, then getting the host header and thus throwing
away the resource config can be official

On Tue, Oct 11, 2016 at 9:11 AM Casey Link <notifications@github.com> wrote:

Yup, access to headers is quite important IMHO. Otherwise we need a
Katharsis Filter to do things w/ katharsis context, and another filter
before Katharsis to do things with the headers.


—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/katharsis-project/katharsis-framework/issues/154#issuecomment-252964970>,
or mute the thread
